litestar/openapi/plugins.py

```diff
@@ -0,0 +1,584 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Sequence
+
+import msgspec
+import yaml
+
+from litestar.enums import MediaType, OpenAPIMediaType
+from litestar.handlers import get
+from litestar.serialization import encode_json, get_serializer
+
+if TYPE_CHECKING:
+    from litestar.connection import Request
+    from litestar.router import Router
+
+__all__ = (
+    "OpenAPIRenderPlugin",
+    "RapidocRenderPlugin",
+    "RedocRenderPlugin",
+    "StoplightRenderPlugin",
+    "SwaggerRenderPlugin",
+    "YamlRenderPlugin",
+)
+
+_favicon_url = "https://raw.githubusercontent.com/litestar-org/branding/main/assets/Branding%20-%20PNG%20-%20Transparent/Badge%20-%20Blue%20and%20Yellow.png"
+_default_favicon = f"<link rel='icon' type='image/png' href='{_favicon_url}'>"
+
+
+class OpenAPIRenderPlugin:
```

Comment:
`OpenAPIUIPlugin`? I realize that I've inadvertently used the phrasing "OpenAPI UI Plugin" throughout the docs - I wonder if that naming is more intuitive?

Reply:
Or just `OpenAPIPlugin`?

Reply:
`OpenAPIRenderPlugin` is a bit verbose but I think that's okay. `OpenAPIPlugin` is too vague, an `OpenAPI UI Plugin` is not an accurate description, since these are used to not only display UIs, but also JSON and YAML.